src/OrleansTestingHost/AppDomainSiloHost.cs

```diff
@@ -12,31 +11,37 @@
 using Orleans.Runtime.GrainDirectory;
 using Orleans.Runtime.Messaging;
 using Orleans.Runtime.Placement;
-using Orleans.Runtime.TestHooks;
 using Orleans.Storage;
-using Orleans.Runtime.MultiClusterNetwork;
 using Orleans.MultiCluster;
+using Orleans.Hosting;
+using Orleans.Runtime.MultiClusterNetwork;
+using Orleans.Runtime.TestHooks;
+using Orleans.Runtime.Providers;
+using Orleans.Runtime.Storage;
 
 namespace Orleans.TestingHost
 {
     /// <summary>Allows programmatically hosting an Orleans silo in the curent app domain, exposing some marshable members via remoting.</summary>
     public class AppDomainSiloHost : MarshalByRefObject
     {
-        private readonly Silo silo;
+        private readonly ISilo silo;
 
         /// <summary>Creates and initializes a silo in the current app domain.</summary>
         /// <param name="name">Name of this silo.</param>
-        /// <param name="siloType">Type of this silo.</param>
+        /// <param name="siloBuilderFactoryType">Type of silo builder factory.</param>
         /// <param name="config">Silo config data to be used for this silo.</param>
-        public AppDomainSiloHost(string name, Silo.SiloType siloType, ClusterConfiguration config)
+        public AppDomainSiloHost(string name, Type siloBuilderFactoryType, ClusterConfiguration config)
         {
-            this.silo = new Silo(name, siloType, config);
-            this.silo.InitializeTestHooksSystemTarget();
+            var builderFactory = (ISiloBuilderFactory)Activator.CreateInstance(siloBuilderFactoryType);
+            ISiloBuilder builder = builderFactory.CreateSiloBuilder(name, config);
+            builder.ConfigureServices((services) => services.AddSingleton<TestHooksSystemTarget>());
```

Comment:
nit: this is not needed in the container... you can just use ActivatorUtilities to resolve one and register it as a sys target when you are initializing it.

Reply:
It is needed in other areas, not just when initialized.